js/network.js

```diff
@@ -27,7 +27,21 @@ async function sign_tx(api, settings){
   var xrp_secret = document.getElementById("xrp_secret");
   var xrp_address = document.getElementById("xrp_address");
   var eth_address = document.getElementById("eth_address")
-
+  var keyPair = {}
+
+  // generate the keyPair to use for signing purposes, either from the private key, or the secret, depending on what was specified
+  if (offline_api.isValidSecret(xrp_secret.value)) {
+    // a secret key was specified
+    keyPair = offline_api.deriveKeypair(xrp_secret.value);
+  } else {
+    // a private key was specified (64 or 66 chars, double-zeroed or not)
+    const public64 = (xrp_secret.length > 64) ? xrp_secret.value.slice(2) : xrp_secret.value;
+    keyPair = {
+        privateKey: xrp_secret.value,
+        publicKey: bytesToHex(secp256k1.keyFromPrivate(public64).getPublic().encodeCompressed()),
```

Comment:
This does not handle the ED25519 case.

Assuming that the private key is 66 characters in length, we can check the first 2 characters to determine this. If these chars equals 'ED' we know we have a ED25519 key and the public key should be extracted like so:

```
bytesToHex(ed25519.keyFromSecret(sec).getPublic())
```